src/compiler/checker.ts

```diff
@@ -24163,6 +24163,11 @@ namespace ts {
                         return propertyType;
                     }
                 }
+                else if (element.name?.kind === SyntaxKind.ComputedPropertyName) {

```

Comment:
The previous case with `!hasNonBindableDynamicName` should handle this case too. I think it needs to be fixed instead of adding another case below it.

(`!hasNonBindableDynamicName` is very confusing to read, so it's easy to misunderstand. I renamed the function and inverted it in #42297)

Reply:
🤔 `hasBindableName(element)` should not hold true when `element.name.expression` is a complex expression like `['a' + 'b']: x => x`, right? So we still need to add another case?

Reply:
Symbol late binding is interesting insofar as it's binding that we do in the checker... I think it's probably fine to expand the field of types (syntaxes?) we consider non-latebindable? I think the issue is less the possibility of binding issues, and more the possibility of circularity issues arising in the resolved types. @rbuckton probably has a more nuanced opinion.